src/Umbraco.Infrastructure/BackgroundJobs/Jobs/WebhookFiring.cs

```diff
@@ -90,7 +89,7 @@ public async Task RunJobAsync()
 
     private async Task<HttpResponseMessage?> SendRequestAsync(IWebhook webhook, string eventName, string? serializedObject, int retryCount, CancellationToken cancellationToken)
     {
-        using var httpClient = new HttpClient();
+        using HttpClient httpClient = _httpClientFactory.CreateClient(Constants.HttpClients.WebhookFiring);
 
         var stringContent = new StringContent(serializedObject ?? string.Empty, Encoding.UTF8, MediaTypeNames.Application.Json);
         stringContent.Headers.TryAddWithoutValidation("Umb-Webhook-Event", eventName);
```

Comment:
The `TryAddWithoutValidation` doesn't overwrite headers that are already configured as `DefaultRequestHeaders`, probably because they are set on the content, instead of the actual `HttpRequestMessage`. And adding non-content headers on the content also isn't allowed in the first place, so I guess that's why they were added without validation...

We could update this to the following (maybe with some additional validation/exception handling for adding the headers):

```c#
var request = new HttpRequestMessage(HttpMethod.Post, webhook.Url)
{
    Version = httpClient.DefaultRequestVersion,
    VersionPolicy = httpClient.DefaultVersionPolicy,
};

// Add headers
request.Headers.TryAddWithoutValidation("Umb-Webhook-Event", eventName);

foreach (KeyValuePair<string, string> header in webhook.Headers)
{
    request.Headers.TryAddWithoutValidation(header.Key, header.Value);
}

// Set content
request.Content = new StringContent(serializedObject ?? string.Empty, Encoding.UTF8, MediaTypeNames.Application.Json);

// Send request
HttpResponseMessage? response = null;
try
{
    response = await httpClient.SendAsync(request, cancellationToken);
}
catch (Exception ex)
{
    _logger.LogError(ex, "Error while sending webhook request for webhook {WebhookKey}.", webhook);
}
```

Reply:
Done 👍